File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/module/ServiceModule.java
package com.benoitquenaudon.tvfoot.red.app.injection.module;

import android.app.AlarmManager;
import android.app.Application;
import android.content.Context;
import com.benoitquenaudon.tvfoot.red.api.TvfootService;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;
import retrofit2.Retrofit;

@Module public class ServiceModule {
  @Provides @Singleton static TvfootService provideTvfootService(Retrofit retrofit) {
    return retrofit.create(TvfootService.class);
  }

  @Provides static AlarmManager provideAlarmManager(Application context) {
    return (AlarmManager) context.getSystemService(Context.ALARM_SERVICE);
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/injection/module/ServiceModule.kt
package com.benoitquenaudon.tvfoot.red.injection.module

import android.app.AlarmManager
import androidx.content.systemService
import com.benoitquenaudon.tvfoot.red.RedApp
import com.benoitquenaudon.tvfoot.red.api.TvfootService
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import javax.inject.Singleton

@Module
object ServiceModule {
  @JvmStatic
  @Provides
  @Singleton
  fun provideTvfootService(retrofit: Retrofit): TvfootService {
    return retrofit.create(TvfootService::class.java)
  }

  @JvmStatic
  @Provides
  fun provideAlarmManager(context: RedApp): AlarmManager {
    return context.systemService()
  }
}
